src/lib/OpenEXR/ImfScanLineInputFile.cpp

```diff
@@ -244,6 +244,8 @@ ScanLineInputFile::setFrameBuffer (const FrameBuffer& frameBuffer)
     std::lock_guard<std::mutex> lock (_data->_mx);
 #endif
     _data->fill_list.clear ();
+    if (_data->singleScan)
```

Comment:
this might be fine for now, but is also bit unsafe in that I believe it might be triggering a memory leak if the internal representation of the decode pipeline will be initialized twice without a destroy between, so while  i think it'd be safer to just do `_data->singleScan.reset();` for now

Reply:
I replaced the `singleScane->first=true` with `singleScan.reset()`, that satisfies the test, too, thanks.

I ran the re-read test in valgrind and it reported that everything was freed, no leak, for what that's worth.